model/src/main/java/com/basistech/rosette/apimodel/Request.java

```diff
@@ -17,17 +17,21 @@
 package com.basistech.rosette.apimodel;
 
 import lombok.AllArgsConstructor;
-import lombok.Data;
+import lombok.Value;
+import lombok.experimental.NonFinal;
+import lombok.experimental.SuperBuilder;
 
 /**
  * Common base class of all requests.
  */
-@Data
+@Value
+@NonFinal
+@SuperBuilder
 @AllArgsConstructor
 public abstract class Request {
 
     /**
      * @return profile ID
      */
-    protected String profileId;
+    String profileId;
```

Comment:
Why? Some test requirement?

Reply:
@Value makes all the fields private final. No need to annotate them otherwise

Reply:
But `@NonFinal` was also added to the class. So the intention is to have a private non-final field?

Reply:
Got it, so `@Value` here makes pretty confusing to me. Single field abstract class. Let's not use it, just put a getter, keep the constructor.